src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -179,10 +170,15 @@ object RuleUtils {
     // If there is no change in source data files, the index can be applied by
     // transformPlanToUseIndexOnlyScan regardless of Hybrid Scan config.
     // This tag should always exist if Hybrid Scan is enabled.
-    lazy val hybridScanRequired =
+    val hybridScanRequired = HyperspaceConf.hybridScanEnabled(spark) &&
       index.getTagValue(getLogicalRelation(plan).get, IndexLogEntryTags.HYBRIDSCAN_REQUIRED).get
 
-    val transformed = if (HyperspaceConf.hybridScanEnabled(spark) && hybridScanRequired) {
+    // In case the index has appended files and/or deleted files which are updated by quick
+    // refresh mode, we should handle them in query time by using Hybrid Scan.
+    // Because quick refresh doesn't update index data, but just metadata in IndexLogEntry.
+    lazy val quickRefreshedIndex = index.hasSourceUpdate
```

Comment:
To me, RuleUtils doesn't need to know how the index metadata was generated (i.e., it just knows how to understand it). So I think we can just name it as `isSourceUpdated`? And you can update the comment that if `isSourceUpdated` is set to true, index data is not up to date with the source data.